src/plugins/deploy/azureDeployPlugin.ts

```diff
@@ -4,11 +4,41 @@ import { FunctionAppService } from "../../services/functionAppService";
 
 export class AzureDeployPlugin {
   public hooks: { [eventName: string]: Promise<any> };
+  public commands: any;
 
   public constructor(private serverless: Serverless, private options: Serverless.Options) {
     this.hooks = {
-      "deploy:deploy": this.deploy.bind(this)
+      "deploy:deploy": this.deploy.bind(this),
+      "deploy:list:list": this.list.bind(this),
     };
+
+    this.commands = {
+      deploy: {
```

Comment:
Does defining these commands mess with the top level core plugin on the CLI ?

Reply:
I was concerned about that, which is why I left out the `lifecycleEvents` (I defined them at first, which caused `deploy` to be triggered twice). I set the plugin to log `this.commands` to see what the object looks like by default, and itame back undefined. In my manual testing, the deployment still works fine. Feel free to double check me on that though.